rmf_fleet_adapter/src/rmf_fleet_adapter/agv/RobotContext.cpp

```diff
@@ -1496,7 +1496,12 @@ void RobotContext::_check_door_supervisor(
 {
   const auto now = std::chrono::steady_clock::now();
   const auto dt = std::chrono::seconds(10);
-  if (_last_active_task_time + dt < now)
+
+  if (_current_task_id.has_value())
+  {
+    _last_active_task_time = now;
+  }
+  else if (_last_active_task_time + dt < now)
```

Comment:
Some style guides may complain if we have an `else if` block without an `else` so better if we can include something like.
```
else
{
  continue;
}

```